internal/services/environment/resource_environment.go

```diff
@@ -590,8 +612,11 @@ func updateExistingDataverse(ctx context.Context, plan *SourceModel, environment
 	plan.Dataverse.As(ctx, &dataverseSourcePlanModel, basetypes.ObjectAsOptions{UnhandledNullAsEmpty: true, UnhandledUnknownAsEmpty: true})
 
 	environmentDto.Properties.LinkedEnvironmentMetadata = &LinkedEnvironmentMetadataDto{
-		SecurityGroupId: dataverseSourcePlanModel.SecurityGroupId.ValueString(),
-		DomainName:      dataverseSourcePlanModel.Domain.ValueString(),
+		DomainName: dataverseSourcePlanModel.Domain.ValueString(),
+	}
+
+	if plan.EnvironmentType.ValueString() != "Developer" {
```

Comment:
does case matter? do we need to be case insensitive?

Reply:
make a const

Reply:
lets make it as conts